openerp/models.py

```diff
@@ -3898,7 +3898,7 @@ def _write(self, cr, user, ids, vals, context=None):
             params = tuple(u[2] for u in updates if len(u) > 2)
             for sub_ids in cr.split_for_in_conditions(ids):
                 cr.execute(query, params + (sub_ids,))
-                if cr.rowcount != len(sub_ids):
+                if cr.rowcount != len(sub_ids) and cr.rowcount != len(set(sub_ids)):
```

Comment:
@simongoffin What about ensuring that `sub_ids` does not contain duplicates?

    for sub_ids in cr.split_for_in_condition(set(ids)):
                                             ^^^